scripts/install_prerequisites.sh

```diff
@@ -340,13 +385,67 @@ if [ -n "$AWS_INSTALL_PREFIX" ] && [ -z "$(echo $exclude_prereq | grep aws)" ];
       -DAUTORUN_UNIT_TESTS=OFF
     cmake --build . --config=Release
     cmake --install . --config=Release
-    cd ../.. && rm -rf 1.11.454.tar.gz aws-sdk-cpp
+    cd ../..
+    rm -rf 1.11.454.tar.gz
+    if [ "$keep_sources" = true ]; then
+      mkdir -p "$tpls_dir"
+      mv aws-sdk-cpp "$tpls_dir"
+    else
+      rm -rf aws-sdk-cpp
+    fi
     remove_temp_installs
   else
     echo "AWS SDK already installed in $AWS_INSTALL_PREFIX."
   fi
 fi
 
+lookup_tpls_sha() {
+  local path="$1"
+
+  # Using lock file
+  if [[ -f /tmp/tpls_commits.lock ]]; then
```

Comment:
This seems rather specific to the nvqc dockerfile. What was the motivation for not just directly using git here (instead of in the dockerfile) to get the shas?